core/server/common/src/main/java/alluxio/master/AbstractPrimarySelector.java

```diff
@@ -67,6 +67,11 @@ public final NodeState getState() {
     }
   }
 
+  @Override
```

Comment:
the only question i'd like to ask is that do we really need a getState() function to read a VOLATILE with the protection of a lock? seems like there isn't any intermediate state of mState set by setState() which can cause issues. wdyt @tcrain @jenoudet

Reply:
You would have to look around at where its used and check if something expects to be blocked while state transition is happening.

Reply:
I find that `volatile` variables are difficult to reason about. I prefer the simplicity and clarity that comes with locks.

Reply:
okay if you are okay with this unsafe() interface, let's just go with it!

Reply:
the lambda in the listener isn't thread safe so i believe adding a lock here is still necessary.